appservice/src/editScmType.ts

```diff
@@ -56,11 +61,25 @@ async function showScmPrompt(currentScmType: string): Promise<string> {
         placeHolder: localize('scmPrompt', 'Select a new source.'),
         suppressPersistence: true
     };
-    const newScmType: string | undefined = (await ext.ui.showQuickPick(scmQuickPicks, options)).data;
+    const newScmType: ScmType | undefined = <ScmType>(await ext.ui.showQuickPick(scmQuickPicks, options)).data;
     if (newScmType === undefined) {
         // if the user clicks the current source, treat it as a cancel
         throw new UserCancelledError();
     } else {
         return newScmType;
     }
 }
+
+export async function detectProjectScmType(fsPath: string): Promise<ScmType> {
```

Comment:
Your gonna have to give me more detail on why you're adding this function. Seems like some magic black-box logic